test/runnable/test16640.d

```diff
@@ -0,0 +1,11 @@
+// PERMUTE_ARGS:
+
+void testFileFullPathAsDefaultArgument(string preBakedFileFullPath, string fileFullPath = __FILE_FULL_PATH__)
+{
+    assert(preBakedFileFullPath == fileFullPath);
+}
+
+void main()
+{
+    testFileFullPathAsDefaultArgument(__FILE_FULL_PATH__);
```

Comment:
The test in line.d checks if `__FILE_FULL_PATH__` ends with the same expansion as `__FILE__` and that it's an absolute path, but doesn't actually check whether the full path is correct. @wilzbach 's idea of having a test pass in the actual full path and comparing it with `__FILE_FULL_PATH__` would be a nice addition, not sure how it would be implemented though.  I wouldn't block this PR without it, but this PR would be even better with a new test that does this.

P.S. I made a mistake in my previous comment about initialization, feel free to ignore it :)